src/main/java/de/medizininformatikinitiative/flare/model/sq/ConceptFilterPart.java

```diff
@@ -35,12 +34,6 @@ ConceptFilterPart appendConcept(TermCode concept) {
 
     @Override
     public Either<Exception, List<ExpandedFilter>> expand(LocalDate today, FilterMapping filterMapping) {
-        return Either.right(concepts.stream()
-                .map(concept -> switch (filterMapping.type()) {
-                    case CODE -> (ExpandedFilter) new ExpandedCodeFilter(filterMapping.searchParameter(),
-                            concept.code());
-                    case CODING -> new ExpandedConceptFilter(filterMapping.searchParameter(), concept);
-                })
-                .toList());
+        return filterMapping.expandConceptFilterPart(concepts);
```

Comment:
I ment, moving the expansion of a single concept into `FilterMapping` and not just the expansion of all the concepts. I liked to simplify the method by splitting up the mapping of a single concept from the handling of all concepts.